src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SafeEvpPKeyHandle.OpenSsl.cs

```diff
@@ -31,13 +38,53 @@ public SafeEvpPKeyHandle(IntPtr handle, bool ownsHandle)
         {
         }
 
+        internal SafeEvpPKeyHandle(IntPtr handle, IntPtr extraHandle)
+            : base(handle, ownsHandle: true)
+        {
+            ExtraHandle = extraHandle;
+        }
+
         protected override bool ReleaseHandle()
         {
-            Interop.Crypto.EvpPkeyDestroy(handle);
+            Interop.Crypto.EvpPkeyDestroy(handle, ExtraHandle);
+            ExtraHandle = IntPtr.Zero;
+
             SetHandle(IntPtr.Zero);
             return true;
         }
 
+        internal static SafeEvpPKeyHandle GenerateRSAKey(int keySize)
```

Comment:
Seems strange to have all these generate APIs dangle off `SafeEvpPKeyHandle`. Doesn't seem like key generation should be a concern of the safe handle.